pkg/sdkstats/sdkstats.go

```diff
@@ -130,33 +139,52 @@ func GetRepoDetails(client *github.Client, owner string, name string) map[string
 	repoMap["CreatedAt"] = repos.GetCreatedAt().String()
 	repoMap["PushedAt"] = repos.GetPushedAt().String()
 	repoMap["TotalCommits"] = strconv.Itoa(total)
-	return repoMap
+	repoMap["Language"] = repos.GetLanguage()
+	return repoMap, nil
 
 }
 
-func getversion(codeResults github.CodeResult, projectType string) int {
+//getVersion takes each Code result from a search output, and searches for Fragments comtaining VERSION, and returns the same.
+//The Fragments Text Macthes used are specific to Operator-SDK Version Pattern.
+func getversion(codeResults github.CodeResult, projectType string) string {
 
-	var searchQ string
-	var SDKversion int
+	var searchQ, SDKversion, searchLatest string
+	SDKLatest := "11"
 
 	if projectType == "helm" {
 		searchQ = "quay.io/operator-framework/helm-operator:v0."
+		searchLatest = "quay.io/operator-framework/helm-operator:latest"
 	}
 	if projectType == "ansible" {
 		searchQ = "quay.io/operator-framework/ansible-operator:v0."
+		searchLatest = "quay.io/operator-framework/ansible-operator:latest"
+
 	}
 	if projectType == "go.mod" {
 		searchQ = "github.com/operator-framework/operator-sdk v0."
 	}
 
-loop:
-	for j := 1; j <= 11; j++ {
-		for _, r := range codeResults.TextMatches {
-			if strings.Contains(r.GetFragment(), searchQ+strconv.Itoa(j)+".") {
-				SDKversion = j
-				break loop
+	for _, r := range codeResults.TextMatches {
+		if strings.Contains(r.GetFragment(), searchQ) {
+			posFirst := strings.Index(r.GetFragment(), searchQ)
+			if posFirst == -1 {
+				SDKversion = "N/A"
 			}
+			posFirstAdjusted := posFirst + len(searchQ)
+			runes := []rune(r.GetFragment())
+			c := posFirstAdjusted + 2
+			if c == -1 {
+				SDKversion = "N/A"
+			}
+			fmt.Println("HERE", string(runes[posFirstAdjusted:c]))
+			SDKversion = strings.Trim(string(runes[posFirstAdjusted:c]), ".")
+			fmt.Println("THERE", SDKversion)
+		} else if (projectType == "helm" || projectType == "ansible") && strings.Contains(r.GetFragment(), searchLatest) {
+			SDKversion = SDKLatest
 		}
 	}
+	if SDKversion == "" {
+		SDKversion = "0"
+	}
```

Comment:
In this function, you also have advance knowledge of the project types. Again, since this is a library, that might not always be the case. A good way to solve this might be to also include an interface in the `RepoMetadataQuery` struct. The the library can just call the version parser without having to know its implementation.

```go
type RepoMetadataQuery struct {
	ProjectType string
	Queries []string
	VersionParser VersionParser
}
```

And then define the following:

```go
// Declare this in pkg/sdkstats/sdkstats.go
type VersionParser interface {
	ParseVersion(codeResults github.CodeResult) (string, error)
}

// Declare the implementations of the interface in main.go
type helmVersionParser struct {}

func (p helmVersionParser) ParseVersion(codeResults github.CodeResult) (string, error) {
	// parse the version, assuming a helm result. if a version isn't found, return an error
}

// Do something similar for ansible and go projects.
type ansibleVersionParser struct {}
type goVersionParser struct {}
```

With this, `queries` in `main.go` would look like this:
```go
queries := []sdkstats.RepoMetadataQuery{
	sdkstats.RepoMetadataQuery{
		ProjectType: "ansible",
		Queries: []string{"ansible-query-string-here"},
		VersionParser: &ansibleVersionParser{}
	},
	sdkstats.RepoMetadataQuery{
		ProjectType: "helm",
		Queries: []string{"helm-query-string-here"},
		VersionParser: &helmVersionParser{}
	},
	sdkstats.RepoMetadataQuery{
		ProjectType: "go",
		Queries: []string{
			"go-mod-query-string-here",
			"gopkg-toml-query-string-here",
		},
		VersionParser: &goVersionParser{}
	},
}
```
```